internal/engine/wazevo/backend/isa/arm64/machine_relocation.go

```diff
@@ -32,3 +39,38 @@ func (m *machine) ResolveRelocations(refToBinaryOffset map[ssa.FuncRef]int, bina
 		}
 	}
 }
+
+func (m *machine) UpdateRelocationInfo(r *backend.RelocationInfo, totalSize int, body []byte) []byte {
+	// FIXME: this should add padding conditionally based on refToBinaryOffset[r.FuncRef].
+	// But when we invoke this method the refToBinaryOffset is not set for all funcRefs.
+	r.Offset += int64(totalSize)
+	r.TrampolineOffset = totalSize + len(body)
+	body = append(body, make([]byte, 4*6)...)
+	return body
+}
+
+func encodeTrampoline(addr uint, binary []byte, instrOffset int, returnOffset int64) {
+	// The tmpReg is safe to overwrite.
+	tmpReg := regNumberInEncoding[tmp]
+
+	const movzOp = uint32(0b10)
+	const movkOp = uint32(0b11)
+	// Note: for our purposes the 64-bit width of the reg should be enough.
+	//       however, larger values could be written to and loaded from a constant pool (see encodeBrTableSequence).
+	instrs := []uint32{
+		encodeMoveWideImmediate(movzOp, tmpReg, uint64(uint16(addr)), 0, 1),
+		encodeMoveWideImmediate(movkOp, tmpReg, uint64(uint16(addr>>16)), 1, 1),
+		encodeMoveWideImmediate(movkOp, tmpReg, uint64(uint16(addr>>32)), 2, 1),
+		encodeMoveWideImmediate(movkOp, tmpReg, uint64(uint16(addr>>48)), 3, 1),
+		encodeUnconditionalBranchReg(tmpReg, true),
+		encodeUnconditionalBranch(false, returnOffset-6*4),
```

Comment:
it's possible that we can resolve the width of the jump earlier and either generate a smaller trampoline, with less instructions, or use a different strategy for encoding the address (for now I just wanted to see if the general approach was working)